mobile/lib/models/kya.dart

```diff
@@ -83,24 +82,40 @@ class Kya extends Equatable {
   }
 
   @override
-  List<Object> get props => [id];
+  List<Object> get props => [id, title, progress];
 }
 
 @JsonSerializable(explicitToJson: true)
 class KyaLesson extends Equatable {
   const KyaLesson({
+    required this.id,
     required this.title,
     required this.imageUrl,
     required this.body,
   });
 
-  factory KyaLesson.fromJson(Map<String, dynamic> json) =>
-      _$KyaLessonFromJson(json);
+  factory KyaLesson.fromJson(Map<String, dynamic> json) {
+    return _$KyaLessonFromJson(json);
+  }
+
+  factory KyaLesson.fromKya(Kya kya, int index) {
+    return KyaLesson(
+      id: kya.lessons[index].id,
+      title: kya.lessons[index].title,
+      imageUrl: kya.lessons[index].imageUrl,
+      body: kya.lessons[index].body,
+    );
+  }
+
+  @JsonKey(name: '_id')
+  final String id;
 
   final String title;
 
+  @JsonKey(name: 'image')
   final String imageUrl;
 
+  @JsonKey(name: 'content')
   final String body;
```

Comment:
I think we can rename body to content. @BenjaminSsempala

Reply:
Done